api/test/wfl/integration/modules/covid_test.clj

```diff
@@ -554,43 +554,96 @@
          (is (== 1 (covid/queue-length! executor)))
          (is (not (covid/done? executor)))))))
 
+(def ^:private fake-entity-type "flowcell")
+(def ^:private fake-entity-name "test")
+
 (deftest test-update-terra-workspace-sink
-  (let [flowcell-id "test"
-        entity-type "flowcell"
-        workflow    {:uuid    "2768b29e-c808-4bd6-a46b-6c94fd2a67aa"
+  (let [workflow    {:uuid    "2768b29e-c808-4bd6-a46b-6c94fd2a67aa"
                      :status  "Succeeded"
                      :outputs (-> "sarscov2_illumina_full/outputs.edn"
                                   resources/read-resource
-                                  (assoc :flowcell_id flowcell-id))}
+                                  (assoc :flowcell_id fake-entity-name))}
         executor    (make-queue-from-list [workflow])
         sink        (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
                       (->> {:name           "Terra Workspace"
                             :workspace      "workspace-ns/workspace-name"
-                            :entityType     entity-type
+                            :entityType     fake-entity-type
                             :fromOutputs    (resources/read-resource
                                              "sarscov2_illumina_full/entity-from-outputs.edn")
                             :identifier     "flowcell_id"
                             :skipValidation true}
                            (covid/create-sink! tx (rand-int 1000000))
                            (zipmap [:sink_type :sink_items])
                            (covid/load-sink! tx)))]
-    (letfn [(verify-upsert-request [workspace [[name type _]]]
+    (letfn [(verify-upsert-request [workspace [[type name _]]]
               (is (= "workspace-ns/workspace-name" workspace))
-              (is (= name flowcell-id))
-              (is (= type entity-type)))]
+              (is (= type fake-entity-type))
+              (is (= name fake-entity-name)))
+            (throw-if-called [fname & args]
```

Comment:
Nice, I was wondering how to achieve something like this in testing, will bank this.